packages/core/src/lib/parse.keypress-kitty.ts

```diff
@@ -222,11 +246,83 @@ function parseKittyFunctionalKey(sequence: string): ParsedKey | null {
   return key
 }
 
+/**
+ * Parse Kitty keyboard protocol tilde keys with event type
+ * Format: CSI number;modifiers:event_type ~
+ * Examples:
+ *   \x1b[5;1:1~ = pageup press
+ *   \x1b[5;1:3~ = pageup release
+ *   \x1b[3;5:1~ = ctrl+delete press
+ */
+function parseKittyTildeKey(sequence: string): ParsedKey | null {
```

Comment:
I think that method can be combined with the functional parsing as it is the same format just with tilde, so the regex can detect both, otherwise it seems like duplicate logic?

Reply:
Definitely! It was easier to reason about separately while iterating but it can be combined